velox/expression/SignatureBinder.cpp

```diff
@@ -369,18 +438,42 @@ bool SignatureBinderBase::tryBind(
           return false;
         }
         break;
-      case TypeParameterKind::kType:
+      case TypeParameterKind::kType: {
         if (!checkNamedRowField(params[i], actualType, i)) {
           return false;
         }
-
-        if (!tryBind(params[i], actualParameter.type)) {
-          // TODO Allow coercions for complex types.
+        Coercion childCoercion;
+        if (!tryBind(
+                params[i],
+                actualParameter.type,
+                coercion ? &childCoercion : nullptr)) {
           return false;
         }
+        if (childCoercion.type) {
```

Comment:
```suggestion
        if (childCoercion.cost > 0) {
```

In such case this condition 
```
    if (coercion->type->equivalent(*actualType)) {
      coercion->reset();
    }
```
is unnecessary